.github/workflows/build.yaml

```diff
@@ -91,8 +91,8 @@ jobs:
           cpack
       - name: Install dev package
         run: |
-          apt update && apt upgrade -y
-          apt install -y ./build/leap_*.deb ./build/leap-dev*.deb
+          apt-get update
```

Comment:
reason for removal of `upgrade`? there was some uncertainly around best practices last time this came up. Not doing the `upgrade` means you could end up with "mixed date" dependency installations: if `foo` depends on `bar`, and `bar` is already installed _from a year ago_, doing `apt update && apt install foo` I believe means the latest version of `foo` will be using a one year old `bar` (unless `foo` has an explicit version dependency)

But, I believe Ubuntu tries to maintain ABI compatibility within a release? That seems potentially hard over the course of 4 years. But maybe.

TLDR it was never clear to me the canonical best practice here; but always doing a `upgrade` seems like the safest option.

Reply:
Yeah it is difficult to say. Your comment had me go refresh on this as well.  My reasoning here was that the `upgrade` step can force one's hand into installing a host of updated packages that aren't being used by our use case, whereas relying on the deb package install to take care of meeting its own dependency tree would be safe -- so long as the package index had been updated.  Anyway, for now I've added it back in and we can punt this down the road for future discussion...or not.